go/teams/service_helper.go

```diff
@@ -31,268 +30,176 @@ func LoadTeamPlusApplicationKeys(ctx context.Context, g *libkb.GlobalContext, id
-		for _, row := range *rows {
-			if row.Status != keybase1.TeamMemberStatus_DELETED {
-				filtered = append(filtered, row)
-			} else {
-				g.Log.CDebugf(ctx, "membersHideDeletedUsers filtered out row: %v %v", row.Uv, row.Status)
-			}
+func membersFilterDeletedUsers(ctx context.Context, g *libkb.GlobalContext, members []keybase1.TeamMemberDetails) (ret []keybase1.TeamMemberDetails) {
+	for _, member := range members {
+		if member.Status != keybase1.TeamMemberStatus_DELETED {
+			ret = append(ret, member)
+		} else {
+			g.Log.CDebugf(ctx, "membersHideDeletedUsers filtered out row: %v %v", member.Uv, member.Status)
 		}
-		*rows = filtered
 	}
+	return ret
 }
 
 // If a UID appears multiple times with different TeamMemberStatus, only show the 'ACTIVE' version.
-// This can happen when an owner resets and is re-added as an admin by an admin.
-// Mutates `members`
-func membersHideInactiveDuplicates(ctx context.Context, g *libkb.GlobalContext, members *keybase1.TeamMembersDetails) {
-	// If a UID appears multiple times with different TeamMemberStatus, only show the 'ACTIVE' version.
-	// This can happen when an owner resets and is re-added as an admin by an admin.
+// This can happen when an owner resets and is re-added by an admin (though the admin could
+// choose to remove the old owner if they so wished).
+func membersHideInactiveDuplicates(ctx context.Context, g *libkb.GlobalContext, members []keybase1.TeamMemberDetails) (ret []keybase1.TeamMemberDetails) {
 	seenActive := make(map[keybase1.UID]bool)
-	lists := []*[]keybase1.TeamMemberDetails{
-		&members.Owners,
-		&members.Admins,
-		&members.Writers,
-		&members.Readers,
-		&members.Bots,
-		&members.RestrictedBots,
-	}
 	// Scan for active rows
-	for _, rows := range lists {
-		for _, row := range *rows {
-			if row.Status == keybase1.TeamMemberStatus_ACTIVE {
-				seenActive[row.Uv.Uid] = true
-			}
+	for _, member := range members {
+		if member.Status == keybase1.TeamMemberStatus_ACTIVE {
+			seenActive[member.Uv.Uid] = true
 		}
 	}
 	// Filter out superseded inactive rows
-	for _, rows := range lists {
-		filtered := []keybase1.TeamMemberDetails{}
-		for _, row := range *rows {
-			if row.Status == keybase1.TeamMemberStatus_ACTIVE || !seenActive[row.Uv.Uid] {
-				filtered = append(filtered, row)
-			} else {
-				g.Log.CDebugf(ctx, "membersHideInactiveDuplicates filtered out row: %v %v", row.Uv, row.Status)
-			}
+	for _, member := range members {
+		if member.Status == keybase1.TeamMemberStatus_ACTIVE || !seenActive[member.Uv.Uid] {
+			ret = append(ret, member)
+		} else {
+			g.Log.CDebugf(ctx, "membersHideInactiveDuplicates filtered out row: %v %v", member.Uv, member.Status)
 		}
-		*rows = filtered
 	}
+	return ret
 }
 
-func MembersDetails(ctx context.Context, g *libkb.GlobalContext, t *Team) (ret keybase1.TeamMembersDetails, err error) {
+func MembersDetails(ctx context.Context, g *libkb.GlobalContext, t *Team) (ret []keybase1.TeamMemberDetails, err error) {
```

Comment:
might as well change it to take mctx while we are there

Reply:
Gave this a shot but its being called from chat where i'd be uncomfortable with making a libkb.NewMetaContext